app/src/lib/git/diff.ts

```diff
@@ -120,8 +124,10 @@ export async function getCommitDiff(
     args.push(file.status.oldPath)
   }
 
+  const argsToUse = args.filter(arg => arg !== '')
```

Comment:
:art: rather than needing to parse like this, you could just conditionally insert the `-w` at the right point:

```diff
diff --git a/app/src/lib/git/diff.ts b/app/src/lib/git/diff.ts
index d33a9620f..ca5d09cc9 100644
--- a/app/src/lib/git/diff.ts
+++ b/app/src/lib/git/diff.ts
@@ -101,22 +101,23 @@ export async function getCommitDiff(
   commitish: string,
   hideWhitespaceInDiff: boolean = false
 ): Promise<IDiff> {
-  const whitespaceArg = hideWhitespaceInDiff ? '-w' : ''
-
   const args = [
     'log',
     commitish,
-    whitespaceArg,
     '-m',
     '-1',
     '--first-parent',
     '--patch-with-raw',
     '-z',
     '--no-color',
-    '--',
-    file.path,
   ]
 
+  if (hideWhitespaceInDiff) {
+    args.push('-w')
+  }
+
+  args.push('--', file.path)
+
   if (
     file.status.kind === AppFileStatusKind.Renamed ||
     file.status.kind === AppFileStatusKind.Copied
@@ -124,10 +125,8 @@ export async function getCommitDiff(
     args.push(file.status.oldPath)
   }
 
-  const argsToUse = args.filter(arg => arg !== '')
-
   const { output } = await spawnAndComplete(
-    argsToUse,
+    args,
     repository.path,
     'getCommitDiff'
   )
```